src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java

```diff
@@ -349,7 +349,7 @@ private void validateEncoder(final KNNMethodContext knnMethodContext, final Vect
                 return;
             }
 
-            if (VectorDataType.BINARY != vectorDataType) {
+            if (VectorDataType.FLOAT == vectorDataType) {
```

Comment:
why this change?

Reply:
Like Binary we are not supporting any encoder for byte vector as we are already using SQ. So, updated the conditions.